### DISCUSSION TITLE
Inference not running on M1 Mac

### DISCUSSION THREAD

**Discussion by phragmosis** [talmolab/sleap#784] [Help!]
Hi, I'm trying to track videos of single fish using an M1 Mac (Monterey v12.4).

The training seems to work well after following the tutorial (although the first model takes about an hour to run 10 epochs). Once training is complete however, I get a pop-up error message before inference starts saying: "An error occcured during inference. Your command line terminal may have more information about the error."

A screenshot of a portion of my terminal shows this error.

I'm afraid I am a complete novice when it comes to Python, so any pointers could be massively appreciated. The main reason I was hoping sleap.ai would work for my project is because my videos unfortunately have strong shadows in them created by the fish (a requirement due to the type of experiment I'm running), so the AI training is pretty vital compared to other tracking algorithms which get confused between a fish and a shadow.

Thank you very much.

<img width="1439" alt="Screenshot 2022-06-09 at 15 12 02" src="https://user-images.githubusercontent.com/20955372/172870447-91ab5c30-9011-4d87-9527-239d3de18d8f.png">

**Comment by talmo** [comment]
Hi @phragmosis,

Thanks for reaching out! Do you mind scrolling up in the error log? The relevant lines are just above where you cut off the screen shot (ending at the line that says `predict_function -> predict_function`).

Thanks!

Talmo

**Reply by phragmosis** [reply]
Whoops, sorry about that @talmo, but thanks for getting back to me so quickly!

I've attached literally everything below where I run the inference predictions separately from the training now.

Many thanks!
Rob

<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173571861-fcec629b-49f6-42a5-9d86-e2e1dc355c15.png">
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173572027-7e26b422-2f11-453a-b8da-7b063472aedb.png">
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173572048-a351e8fc-7901-49a5-8366-0a13564eb929.png">
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173572073-f2505cb5-32f2-4751-899e-6bb66ad9940e.png">
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173572089-6ed2d223-13b7-47f9-a9fb-4e932f85076b.png">
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/20955372/173572111-14cc848c-acfd-413f-8ec4-574f96fa1a9a.png">

**Reply by talmo** [reply]
Oo, I see, strange but I think it might be related to how we preprocess videos during inference. We have a fix that might not be in the M1-compatible branch yet, but let's try it out to be sure. Do you mind sharing your models + sample video with us so we can test it out and see if we can reproduce it on our end?

You can email a Google Drive link or something to `talmo@salk.edu`.

**Reply by phragmosis** [reply]
Thanks very much for your help @talmo. Have emailed you.
